lab-katherine/lib/aws-s3.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+const debug = require('debug')('cfgram:aws-s3')
+
+const AWS = require('aws-sdk')
+const s3 = new AWS.S3()
+s3.config.setPromisesDependency(require('bluebird'))
+
+module.exports = function(params) {
+  debug('#s3Upload')
+
+  return new Promise((resolve, reject) => {
+    s3.upload(params, (err, s3Data) => {
+      if(err) reject(err)
+      resolve(s3Data)
+    })
+    s3.findByIdAndRemove(params, (err, s3Data) => {
```

Comment:
findByIdAndRemove is a mongoose method so it wont work when used on the s3 object.